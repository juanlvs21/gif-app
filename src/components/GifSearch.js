import React, { useState } from "react";
import PropTypes from "prop-types";

const GifSearch = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleChangeInput = ({ target: { value } }) => setInputValue(value);

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 0) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleChangeInput} />
    </form>
  );
};

GifSearch.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default GifSearch;
