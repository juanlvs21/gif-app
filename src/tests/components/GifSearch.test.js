import { shallow } from "enzyme";

import GifSearch from "../../components/GifSearch";

describe("Pruebas en el componente GifSearch", () => {
  const setCategories = jest.fn();

  let wrapper = shallow(<GifSearch setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();

    wrapper = shallow(<GifSearch setCategories={setCategories} />);
  });

  test("Cargar componente GifSearch correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe cambiar el input", () => {
    const input = wrapper.find("input");
    const value = "Example test";

    input.simulate("change", { target: { value } });
  });

  test("Submit: input no cumple la condiciÃ³n inputValue.trim().length > 0", () => {
    const form = wrapper.find("form");
    form.simulate("submit", { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Submit: input con valores validos", () => {
    // const form = wrapper.find("form");
    const value = "Example test";
    wrapper.find("input").simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(wrapper.find("input").prop("value").trim()).toBe("");
  });
});
