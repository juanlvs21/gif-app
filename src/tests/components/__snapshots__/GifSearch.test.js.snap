// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifSearch Cargar componente GifSearch correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
